frontend/src/actions/visualization.js

```diff
@@ -0,0 +1,120 @@
+import { V2 } from "../openapi";
+import { handleErrors } from "./common";
+import config from "../app.config";
+import { getHeader } from "../utils/common";
+
+export const GET_VIZ_CONFIG = "GET_VIZ_CONFIG";
+
+export function getVizConfig(resourceId) {
+	return (dispatch) => {
+		return V2.VisualizationsService.getResourceVizconfigApiV2VisualizationsResourceIdConfigGet(
+			resourceId
+		)
+			.then((json) => {
+				dispatch({
+					type: GET_VIZ_CONFIG,
+					receivedAt: Date.now(),
+					vizConfig: json,
+				});
+			})
+			.catch((reason) => {
+				dispatch(handleErrors(reason, getVizConfig(resourceId)));
+			});
+	};
+}
+
+export const GET_VIZ_DATA = "GET_VIZ_DATA";
+
+export function getVizData(visualizationId) {
+	return (dispatch) => {
+		return V2.VisualizationsService.getVisualizationApiV2VisualizationsVisualizationIdGet(
+			visualizationId
+		)
+			.then((json) => {
+				dispatch({
+					type: GET_VIZ_DATA,
+					receivedAt: Date.now(),
+					vizData: json,
+				});
+			})
+			.catch((reason) => {
+				dispatch(handleErrors(reason, getVizData(visualizationId)));
+			});
+	};
+}
+
+export const DOWNLOAD_VIZ_DATA = "DOWNLOAD_VIZ_DATA";
+
+export function downloadVizData(
+	visualizationId,
+	filename = "",
+	autoSave = true
+) {
+	return async (dispatch) => {
+		if (filename === "") {
+			// TODO guess extension
+			filename = `${visualizationId}.tmp`;
+		}
+		let endpoint = `${config.hostname}/api/v2/visualizations/download/${visualizationId}`;
+		const response = await fetch(endpoint, {
+			method: "GET",
+			mode: "cors",
+			headers: await getHeader(),
+		});
+
+		if (response.status === 200) {
+			const blob = await response.blob();
+			if (autoSave) {
```

Comment:
How is this used?

Reply:
This `autoSave` is just reserved for future usage. I did similar things for download files. 
If autoSave is turned on, frontend/browser will save the blob. It can be used when clicking a link/button then download/save starts.
But for rendering the bytes on browser we just need the blob; hence autoSave should be set to false.